### PR TITLE
Phenomodels lock 'n fold

### LINKED ISSUE #2236: Possibility to lock phenomodels
User request: a function to ‘lock’ the editing of panels to specific staff members.

### LINKED ISSUE #2237: Make main HPO levels foldable in phenomodels
User request: let users expand the upper HPO terms if they want to see the nested terms

### PR DESCRIPTION
**Pull request by northwestwitch**
This PR adds 2 functionalities:
- Lock phenomodels so that they are editable by one or more selected user collaborators --> fix #2236 
- Make the phenomodel checkboxes which have children term foldable (folded since the beginning) --> fix #2237 

**How to test --> local instance of Scout**:
1. create a new user for cust000:
```scout --demo load user -i cust000 -u "Diana Prince" -m wonderwoman@mail.com```
1. Log in as demo user Clark Kent
1. Go to the phenomodels page and create 3 phenomodels with submodels
1. Lock one model with admins Clark Kent and Diana Prince. (model 1)
1. Lock one model with admin Clark Kent (model 2)
1. Don't lock a model at all. (model 3)
1. Log out and log in with user wonderwoman@mail.com
1. Make sure that this user can access all 3 models but can modify only model 1 and 3, including changing the admin and locking these models.
1. Make sure that the phenomodel preview contains foldable divs for the highest level checkboxes which contain children terms (fold and unfold by clicking on their names):
image

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
- [ ] code approved by
- [ ] tests executed by

### CODE REVIEW

**Comment by northwestwitch** [comment]
I'll write a couple of tests tomorrow!

**Comment by northwestwitch** [comment]
Ready for review! Weird that coverage went down on files I didn't touch at all in this PR.

File: scout/server/blueprints/institutes/templates/overview/phenomodels.html [hunk 1]
```diff
@@ -44,6 +45,38 @@
 </div><!-- end of container-float-->
 {% endblock %}
 
+
+{% macro lock_model() %}
```
Comment (dnil) [inline scout/server/blueprints/institutes/templates/overview/phenomodels.html:49]: 👍

File: scout/server/blueprints/institutes/templates/overview/phenomodels.html [hunk 2]
```diff
@@ -108,9 +143,15 @@ <h5 class="card-title">Create new phenotype model</h5>
               <td>{{model.subpanels|length if model.subpanels else '0'}}</td>
               <td>{{model.created.strftime('%Y-%m-%d')}}</td>
               <td>{{model.updated.strftime('%Y-%m-%d')}}</td>
+              <td style="width:30%">{{model.admins|join(", ") or "-"}}</th>
               <td>
                 <button type="button" class="btn btn-danger btn-sm removeModel" data-id="{{model._id}}" aria-hidden="true" data-toggle="modal" data-target="#deleteModal"><i class="fa fa-trash-o"></i></button>
```
Comment (dnil) [inline scout/server/blueprints/institutes/templates/overview/phenomodels.html:148]: Hm, something goes wrong there - perhaps string to ObjectId conversion?

File: scout/server/blueprints/institutes/templates/overview/phenomodels.html [hunk 3]
```diff
@@ -122,10 +163,15 @@ <h5 class="card-title">Create new phenotype model</h5>
 
 {% block scripts %}
 {{ super() }}
+<script src="https://cdn.jsdelivr.net/npm/bootstrap-select@1.13.9/dist/js/bootstrap-select.min.js"></script>
 <script>
   $(document).on("click", ".removeModel", function () {
     var modelId = $(this).data('id');
     $('input#model_id').val(modelId);
 });
+$('select[multiple]').selectpicker({
```
Comment (dnil) [inline scout/server/blueprints/institutes/templates/overview/phenomodels.html:172]: Not in like css? Ok either way I guess.

File: tests/server/blueprints/institutes/test_institute_views.py [hunk 4]
```diff
@@ -83,6 +84,66 @@ def test_phenomodel_GET(app, user_obj, institute_obj):
         assert "Test model" in str(resp.data)
 
 
+def test_phenomodel_lock(app, user_obj, institute_obj):
+    """Test the endpoint to lock a phenomodel and make it editable only by admins"""
```
Comment (dnil) [inline tests/server/blueprints/institutes/test_institute_views.py:88]: 👍

**Review by dnil — APPROVED** [review]
This looks good, but does have a crash on removing phenomodels? I get the feeling it was perhaps there before as well, but hiding. Anyway, decide for yourself if you want to fix it in this or make a new issue!

Screenshot 2021-01-14 at 15 51 24
Screenshot 2021-01-14 at 15 56 04
Screenshot 2021-01-14 at 15 55 15
Screenshot 2021-01-14 at 15 54 44
Screenshot 2021-01-14 at 15 54 05

**Comment by northwestwitch** [comment]
> This looks good, but does have a crash on removing phenomodels? I get the feeling it was perhaps there before as well, but hiding. Anyway, decide for yourself if you want to fix it in this or make a new issue!

I'll check, thanks!

File: scout/server/blueprints/institutes/templates/overview/phenomodels.html (hunk 3 above)
Comment (northwestwitch) [reply scout/server/blueprints/institutes/templates/overview/phenomodels.html:174]: Bah it's quite easy to have it this way..

Comment (northwestwitch) [reply scout/server/blueprints/institutes/templates/overview/phenomodels.html:174]: But there is actually an error in the code, too many }); 
Good that you pointed it out!

**Comment by northwestwitch** [comment]
The error is fixed and I'm going to merge as soon as the tests pass. Thanks for the review and noticing the bug @dnil 🙏

**Pull request merged**